extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/contract_agreements/services/ContractAgreementDataFetcher.java

```diff
@@ -51,21 +57,33 @@ public List<ContractAgreementData> getContractAgreements() {
         var assets = MapUtils.associateBy(getAllAssets(), Asset::getId);
 
         var negotiations = getAllContractNegotiations().stream()
-                .filter(it -> it.getContractAgreement() != null)
-                .collect(groupingBy(it -> it.getContractAgreement().getId()));
+            .filter(it -> it.getContractAgreement() != null)
+            .collect(groupingBy(it -> it.getContractAgreement().getId()));
 
         var transfers = getAllTransferProcesses().stream()
-                .collect(groupingBy(it -> it.getDataRequest().getContractId()));
+            .collect(groupingBy(it -> it.getDataRequest().getContractId()));
+
+        var agreementIds = agreements.stream().map(ContractAgreement::getId).toList();
+
+        var t = SOVITY_CONTRACT_TERMINATION;
+        var terminations = dsl.get().select()
+            .from(t)
+            .where(t.CONTRACT_AGREEMENT_ID.in(agreementIds))
+            .fetch()
+            .into(t)
+            .stream()
+            .collect(Collectors.toMap(SovityContractTerminationRecord::getContractAgreementId, it -> it));
```

Comment:
extract a private method at least

Reply:
structure code with functions. less lines is better. this obviously just wants to do one thing, fetch the terminations, so it's better to move it out of the way for being able to read the entire method faster

Reply:
ah, ok, it as for the whole thing.